tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/UsageReport/WriteUsageBurndownData.cs

```diff
@@ -0,0 +1,124 @@
+            Log.LogMessage(MessageImportance.High, "Generating summary usage burndown data...");
+
+
+            var data = ExecuteGitCommand(RootDirectory, gitLogCommand).AsParallel().Select(commitLine =>
+            {
+                var splitLine = commitLine.Split(',');
+                var commit = new commit() { sha = splitLine[0], title = splitLine[1], commitDate = DateTime.Parse(splitLine[2]) };
+                var fileContents = GetFileContents(baselineRelativeFileName, commit.sha);
+                var usages = UsageData.Parse(XElement.Parse(fileContents)).Usages.NullAsEmpty().ToArray();
+                commit.packageVersionCount = usages.Count();
+                commit.packageCount = usages.GroupBy(i => i.PackageIdentity.Id).Select(grp => grp.First()).Count();
+                return commit;
+            })
+            .Select(c => c.ToString())
+            .Aggregate((a, b) => a + '\n' + b);
+    
+            Directory.CreateDirectory(Path.GetDirectoryName(OutputFilePath));
+
+            File.WriteAllText(OutputFilePath, data);
+
+            Log.LogMessage(
+                MessageImportance.High,
+                $"Generating summary usage burndown data at {OutputFilePath} done. Took {DateTime.Now - startTime}");
+
+            return !Log.HasLoggedErrors;
+        }
+
+        /// <summary>
+        /// Get the contents of a git file based on the commit sha.
+        /// </summary>
+        /// <param name="relativeFilePath">The relative path (from the git root) to the file.</param>
+        /// <param name="commitSha">The commit sha for the version of the file to get.</param>
+        /// <returns>The contents of the specified file.</returns>
+        public string GetFileContents(string relativeFilePath, string commitSha)
+        {
+            WebClient client = new WebClient();
+            var xmlString = client.DownloadString(String.Format("https://raw.githubusercontent.com/dotnet/source-build/{0}/{1}", commitSha, relativeFilePath.Replace('\\', '/')));
+            return xmlString;
+        }
+
+        /// <summary>
+        /// Executes a git command and returns the result.
+        /// </summary>
+        /// <param name="workingDirectory">The working directory for the git command.</param>
+        /// <param name="command">The git command to execute.</param>
+        /// <returns>An array of the output lines of the git command.</returns>
+        public string[] ExecuteGitCommand(string workingDirectory, string command)
+        {
+            Process _process = new Process();
+            _process.StartInfo.FileName = "git";
+            _process.StartInfo.Arguments = command;
+            _process.StartInfo.WorkingDirectory = workingDirectory;
+            _process.StartInfo.RedirectStandardOutput = true;
+            _process.StartInfo.UseShellExecute = false;
+            _process.Start();
+            _process.WaitForExit();
+            return _process.StandardOutput.ReadToEnd().Split('\n');
```

Comment:
Here's info on the potential hang I mentioned: https://stackoverflow.com/q/139593

Reply:
Thanks.  I changed this per the suggestion in the post.